### ISSUE TITLE
_events is shared across subclasses

### ISSUE THREAD

**Issue by lpstein** [Gozala/events#10]
Behavior broke in node 0.10, see https://github.com/joyent/node/issues/7157 ([fixed in node.js master](https://github.com/vkurchatkin/node/commit/2956f0335cfdcd3a8bcefbec73ef194d9eab6f56)).

Example:

``` javascript
var A = function() {};
A.prototype = new events.EventEmitter();

var a1 = new A();
var a2 = new A();

a1.on('test', function() {
  console.log('test');
});
a2.emit('test');
```

This logs 'test' where it shouldn't.

**Comment by darkyen** [comment]
Drop IE 8 ? Please ?

**Comment by leewaygroups** [comment]
@lpstein Looking at your code example, `a2.emit('test');` , logging 'test' is normal.

**Reason**
`A.prototype = new events.EventEmitter();` implies that every instance of A will share among other things same copy of events collection  -- Pure case of prototypal inheritance.

Suppose you do not want this, you can make instances have their own private events collection by:
`var A = function() {`
    `events.EventEmitter.call(this);`
`};`

Note: this does not take away the shared events collection as explained earlier, it only shadows it.

Another option is to require util module then do this:
`util.inherits(A, events.EventEmitter);`


Cheers!!

**Comment by patrickroberts** [comment]
You can also do something like:

    function A() {
      events.EventEmitter.call(this);
    }

    A.prototype = Object.create(events.EventEmitter.prototype);
    A.prototype.constructor = A;

That's one of the methods Babel uses to transpile ES6 inheritance when `Object.setPrototypeOf` doesn't exist. And it doesn't just shadow it, it doesn't even create a shared events collection since `Object.create()` does not invoke the constructor!

**Comment by lpstein** [comment]
This is an automatically generated message.

lpstein@yahoo-inc.com is no longer with Yahoo! Inc.

Your message will not be forwarded.

If you have a sales inquiry, please email yahoosales@yahoo-inc.com and someone will follow up with you shortly.

If you require assistance with a legal matter, please send a message to legal-notices@yahoo-inc.com

Thank you!

**Comment by darkyen** [comment]
Yahoooo spam!

**Comment by goto-bus-stop** [comment]
Fixed in #40
